astropy/convolution/utils.py

```diff
@@ -202,29 +202,25 @@ def discretize_model(model, x_range, y_range=None, mode="center", factor=10):
     if ndim == 1 and y_range is not None:
         raise ValueError("y_range should not be input for a 1D model")
 
-    result = None
```

Comment:
removing this statement is independent of using pattern-matching: `result` is rebinded in all cases, except when it an exception is raised and it doesn't matter anyway.